erpnext/payroll/doctype/employee_benefit_application/employee_benefit_application.js

```diff
@@ -82,14 +100,19 @@ var calculate_all = function(doc) {
 	var tbl = doc.employee_benefits || [];
 	var pro_rata_dispensed_amount = 0;
 	var total_amount = 0;
-	for(var i = 0; i < tbl.length; i++){
-		if(cint(tbl[i].amount) > 0) {
-			total_amount += flt(tbl[i].amount);
-		}
-		if(tbl[i].pay_against_benefit_claim != 1){
-			pro_rata_dispensed_amount += flt(tbl[i].amount);
+	if (doc.max_benefits == 0) {
```

Comment:
all business logic must be server side

https://github.com/frappe/erpnext/wiki/Coding-Standards#all-business-logic-must-be-api-friendly-and-must-be-implemented-in-server-side-python